// Specifies that we are using the proto3 syntax, which is the modern version.
syntax = "proto3";

// This is the definition of our chat message structure.
// It's like defining a class or a struct.
message ChatMessage {
  // Field 1: The ID or username of the sender (still visible for routing).
  string sender_id = 1;

  // Field 2: The encrypted content of the message (was plaintext before).
  bytes encrypted_content = 2;
  
  // Field 3: The nonce/IV used for encryption (needed for decryption).
  bytes nonce = 3;

  // Field 4: A timestamp for when the message was sent.
  int64 timestamp_unix = 4;
}
